app/indexers/hyrax/file_set_indexer.rb

```diff
@@ -62,5 +64,13 @@ def file_format
           object.format_label
         end
       end
+
+      def human_readable_label_name(parent)
+        return unless parent
+
+        work_title = parent.member_of.first&.title&.first
+        page_number = parent.title.first.match(/Page \d+/)&.[](0)
```

Comment:
If we don't have a page number, it seems like we shouldn't check the work_title (which could be expensive)

Consider some short-circuiting logic instead.